joomla/joomla-cms
# Rename incorrectly cased files on updates or when fixing database

Pull Request for Issues https://github.com/joomla/joomla-cms/issues/23725 and  https://github.com/joomla/joomla-cms/issues/30736 .

Replaces Pull Request #30802 .

### Summary of Changes

This pull request (PR) attempts to implement case sensitive file rename and rename some incorrectly cased files.

It is based on PR #30802 plus my corrections for OSX.

Thanks to @SharkyKZ for the original PR #30802.

### Testing Instructions

Testers please report back which server operating system (Linux, Windows, OSX) you have used for your test. If you have several available, test with all of them.

Install Joomla! <= 3.9.23 => check the following files: https://github.com/joomla/joomla-cms/blob/f3666c8062d4dd7f4b204cda28967bba6ed7f04f/administrator/components/com_admin/script.php#L2670-L2674
Use the [update package built by drone for this PR](https://ci.joomla.org/artifacts/joomla/joomla-cms/staging/32006/downloads/39020/Joomla_3.9.24-dev+pr.32006-Development-Update_Package.zip) for Update
- Update with the Joomla Updater 
- Copy the files via FTP and use "Fix Database"
- Install the [installation package built by drone for this PR](https://ci.joomla.org/artifacts/joomla/joomla-cms/staging/32006/downloads/39020/Joomla_3.9.24-dev+pr.32006-Development-Full_Package.zip) as a fresh installation

Now check the file list again => files should in all three test cases have the correct name.

### Actual result BEFORE applying this Pull Request

Incorrectly cased files present.

### Expected result AFTER applying this Pull Request

No incorrectly files after update or database fix or new installation.

### Documentation Changes Required

None.
